File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int left = 0;
        int right =0;
        int i =0;
        int[] merged = new int[m+ n];

        while(left < m && right < n){
            if(nums1[left] < nums2[right]){
                merged[i++] = nums1[left];
                left++;
            }else{
                merged[i++] = nums2[right];
                right++;
            }
        }

        while(left < m){
            merged[i++] = nums1[left];
            left++;
        }

        while(right < n){
            merged[i++] = nums2[right];
            right++;
        }

        for(int k=0;k<m+n;k++){
            nums1[k] = merged[k];
        }

    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        
        i = 0
        j = 0

        lists = []

        while i < m and j < n:
            if nums1[i] > nums2[j]:
               lists.append(nums2[j])
               j += 1
            else:
                lists.append(nums1[i])  
                i += 1 

        while i < m:
            lists.append(nums1[i])
            i += 1

        while j < n:
            lists.append(nums2[j])
            j += 1

        for i in range(m+n):
            nums1[i] = lists[i]       
